tachyon/zk/plonk/circuit/examples/simple_circuit.h

```diff
@@ -31,6 +25,9 @@ class FieldConfig {
   using Field = F;
 
   FieldConfig() = default;
+  FieldConfig(std::array<AdviceColumnKey, 2>&& advice,
+              const InstanceColumnKey& instance, const Selector& s_mul)
+      : advice_(std::move(advice)), instance_(instance), s_mul_(s_mul) {}
```

Comment:
Why did you make it public?

Reply:
Having `FieldChip` and `FieldConfig` constructors private, I had trouble having `SimpleCircuit` being templatized with `FloorPlanner` since there should be friend declaration, which requires, FloorPlanner type as template argument.